File: Week2/김다빈/BOJ_15721_2/15721.py
# A <= 2000 T <= 10000
def manageIndex(idx, a):
    idx %= a
    return idx


def isFinished(sign, t, bCount, dCount):
    if sign == 0:
        if t == bCount:
            return True
        else:
            return False
    else:
        if t == dCount:
            return True
        else:
            return False


def solution():
    a = int(input())
    t = int(input())
    sign = int(input())
    startIdx = 0
    gameRound = 0
    bCount = 0
    dCount = 0
    while True:
        for i in range(4):
            if i % 2 == 0:
                bCount += 1
            else:
                dCount += 1
            if isFinished(sign, t, bCount, dCount):
                return startIdx
            startIdx += 1
            if startIdx >= a:
                startIdx = manageIndex(startIdx, a)


        for i in range(2 * (gameRound + 2)):
            if i < (2 * (gameRound + 2)) // 2:
                bCount += 1
            else:
                dCount += 1
            if isFinished(sign, t, bCount, dCount):
                return startIdx
            startIdx += 1
            if startIdx >= a:
                startIdx = manageIndex(startIdx, a)

        gameRound += 1


print(solution())

File: Week2/김다빈/BOJ_15721_2/15721.java
import java.io.*;

public class Main {
    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    public static int manageIndex(int idx, int a) { return idx % a;}
    public static boolean isFinished(int sign, int t, int bCount, int dCount ) {
        if (sign == 0) {
            return t == bCount;
        } else {
            return t == dCount;
        }
    }

    public static void main(String[] args) throws Exception {
        int a = Integer.parseInt(br.readLine());
        int t = Integer.parseInt(br.readLine());
        int sign = Integer.parseInt(br.readLine());
        int startIdx = 0;
        int gameRound = 0;
        int bCount = 0;
        int dcount = 0;

        while (true) {
            for (int i = 0; i < 4; i ++) {
                if (i % 2 == 0) {
                    bCount++;
                } else {
                    dcount++;
                }

                if (isFinished(sign, t, bCount, dcount)) {
                    startIdx = manageIndex(startIdx, a);
                    System.out.println(startIdx);
                    return;
                } else {
                    startIdx++;
                }
            }

            for (int i = 0; i < 2 * (gameRound + 2); i ++) {
                if (i < (2 * (gameRound + 2)) / 2) {
                    bCount++;
                } else {
                    dcount++;
                }

                if (isFinished(sign, t, bCount, dcount)) {
                    startIdx = manageIndex(startIdx, a);
                    System.out.println(startIdx);
                    return;
                } else {
                    startIdx++;
                }
            }

            gameRound++;
        }
    }
}